src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/EligibilityAndSuitabilityService.kt

```diff
@@ -96,109 +71,33 @@ class EligibilityAndSuitabilityService(
+    val criterionType = CriterionType.valueOf(answer.type.name)
+    val currentAssessment = assessmentService.getCurrentAssessment(prisonNumber)
 
-    val criteria = when {
-      existingCriterionResult != null -> {
-        currentResults.remove(existingCriterionResult)
-        existingCriterionResult.copy(
-          criterionMet = criterion.isMet(answer.answers),
-          questionAnswers = answer.answers,
-          lastUpdatedTimestamp = LocalDateTime.now(),
-        )
-      }
+    with(currentAssessment) {
+      val criterion = policyService.getCriterion(assessmentEntity.policyVersion, criterionType, answer.code)
+      val criterionMet = criterion.isMet(answer.answers)
 
-      else -> EligibilityCheckResult(
-        assessment = assessment,
-        criterionMet = criterion.isMet(answer.answers),
-        questionAnswers = answer.answers,
-        criterionCode = answer.code,
-        criterionVersion = assessment.policyVersion,
-        criterionType = criterionType,
-      )
+      assessmentEntity.addOrReplaceEligibilityCriterionResult(criterionType, criterion.code, criterionMet, answer.answers)
+      assessmentEntity.changeStatus(assessmentLifecycleService.eligibilityAnswerSubmitted(currentAssessment))
+      assessmentRepository.save(assessmentEntity)
     }
-    currentResults.add(criteria)
-    assessmentRepository.save(assessment)
-  }
-
-  /**
-   * Combines eligibility criteria from the policy with the checks on that criteria for a given case
-   */
-  private fun Assessment.getEligibilityProgress(): List<EligibilityCriterionProgress> {
-    val policy = policyService.getVersionFromPolicy(this.policyVersion)
-    val codeToChecks = this.eligibilityCheckResults
-      .filter { it.criterionType == ELIGIBILITY }
-      .associateBy { it.criterionCode }
-
-    return policy.eligibilityCriteria.map { it.toEligibilityCriterionProgress(codeToChecks[it.code]) }
   }
 
-  private fun Criterion.toEligibilityCriterionProgress(eligibilityCheckResult: EligibilityCheckResult?) =
-    EligibilityCriterionProgress(
-      code = code,
-      taskName = name,
-      status = eligibilityCheckResult.getEligibilityStatus(),
-      questions = questions.map {
-        Question(
-          text = it.text,
-          hint = it.hint,
-          name = it.name,
-          answer = eligibilityCheckResult.getAnswer(it.name),
-        )
-      },
+  private fun createAssessmentSummary(
+    assessmentWithEligibilityProgress: AssessmentWithEligibilityProgress,
+  ) = with(assessmentWithEligibilityProgress) {
+    AssessmentSummary(
+      forename = offender.forename,
+      surname = offender.surname,
+      dateOfBirth = offender.dateOfBirth,
+      prisonNumber = offender.prisonNumber,
+      hdced = offender.hdced,
+      crd = offender.crd,
+      location = prison,
+      status = assessmentEntity.status,
+      policyVersion = assessmentEntity.policyVersion,
     )
```

Comment:
This is a lot simpler now a load of the code has been extracted out into the `AssessmentService`